hilti/runtime/include/json.h

```diff
@@ -10,7 +10,7 @@
 #pragma GCC diagnostic push
 #endif
 
-#include <nlohmann/json.hpp>
+#include <hilti/rt/3rdparty/nlohmann/json.hpp>
```

Comment:
Shouldn't this include the header from `3rdparty/json/include/nlohmann` instead?

More generally, I think directly including with our own 3rdparty headers makes this harder to repackage. Ideally We would set up CMake targets in `3rdparty/CMakeLists.txt` or directly `add_subdirectory` to use their predefined targets, and then make our own targets depend on them. This works for both compiled and header-only dependencies, and is exactly what we did for `json` before the changes here.

One benefit of that approach is that building against system-provided dependencies becomes much easier (patch CMake setup instead of patching a number of files). While I suspect right now nobody is interested in that, being able to unbundle dependencies is pretty much required to get into distributions via official package repos.

Reply:
As we discussed, I'm going to leave this as it is. The `3rdparty` includes are now used only for functionality that is needed by the public runtime headers, and they are installed alongside those. That way we make sure the use our own versions, and not accidentally any system-wide installed versions. We could do this through CMake magic as well, but not sure it's worth the trouble right now given that usage is contained to just a handful of places.